sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -69,6 +69,39 @@ object CosmosAccountConfig {
 
 case class CosmosContainerConfig(database: String, container: String)
 
+case class CosmosWriteConfig(upsertEnabled: Boolean, maxRetryCount: Int)
+
+object CosmosWriteConfig {
+  val upsertEnabled = CosmosConfigEntry[Boolean](key = "spark.cosmos.write.upsertEnabled",
```

Comment:
As discussed offline I would prefer having soemthing like

"spark.cosmos.write.mode" or "spark.cosmos.write.strategy" here
with values
- Overwrite (using upserts)
- Append (using create, ignore 409)
- Merge (using patch or create)
- MAYBE Create or CreateOnly or FailOnExisting (using create and fail on 409)

I would prefer that over having upsertEnabled, patchEnabled etc. to express what is happening because I think it is easier to understand.

Reply:
sure these config names look better, will do in the next commit. thanks.

Reply:
done